### PR TITLE
FINERACT-1645:Invalid Balance after Interest posting with withdrawal fee

### PR DESCRIPTION
**Pull request by logoutdhaval**
## Description
 [FINERACT-1645](https://issues.apache.org/jira/browse/FINERACT-1645).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [X] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [X] Create/update unit or integration tests for verifying the changes made.

- [X] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [X] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java [hunk 1]
```diff
@@ -155,7 +155,7 @@ public boolean isDebit() {
     }
 
     public boolean isWithdrawalFeeAndNotReversed() {
-        return this.transactionType.isWithdrawalFee() && isNotReversed();
+        return this.transactionType.isFeeDeduction() && isNotReversed();
```
Comment (fynmanoj) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java:158]: is reversal case already handled?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountInterestPostingServiceImpl.java [hunk 2]
```diff
@@ -449,7 +449,9 @@ protected void recalculateDailyBalances(final Money openingAccountBalance, final
                 }
 
                 runningBalance = runningBalance.plus(transactionAmount);
-                transaction.updateRunningBalance(runningBalance);
+                if (!transaction.getRunningBalance(transactionAmount.getCurrency()).isEqualTo(transactionAmount)) {
```
Comment (fynmanoj) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountInterestPostingServiceImpl.java]: how can we compare transaction amount with running balance? what is the logic here?

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java (hunk 1 above)
Comment (logoutdhaval) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java:158]: yes, that was handled in SavingsAccountTransactionSummaryWrapper.java

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountInterestPostingServiceImpl.java (hunk 2 above)
Comment (logoutdhaval) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountInterestPostingServiceImpl.java]: Sorry, forgot to remove this thing.

**Pull request merged**